Nat as a concrete coercer and abstract validator

erights opened the pull request.

erights commented:
Hi @michaelfig @katelynsills This is now ready for review.

erights commented:
CI seems stuck. I have no idea how to wake it up.

katelynsills commented:
> CI seems stuck. I have no idea how to wake it up.

The CI was never transferred from CircleCI (our old provider) to Github Actions, since we hadn't done work on this repo in a year or so. I can move us over to Github Actions

katelynsills commented:
@erights I made PRs to use Github Actions, yarn, and AVA rather than CircleCI, npm, and tape. Could you look at #144, #145, and #148, and when we are ready and those are approved and merged, rebase on top of master?

katelynsills commented on README.md:
```diff
@@ -4,69 +4,104 @@
+  are JS safe integers, then this accurately represents the mathematical sum
+  of the mathematical numbers they represent.
+
+The bigint datatype, by contrast, is inherently safe. Every bigints `>= 0n`
```
```suggestion
The bigint datatype, by contrast, is inherently safe. Every bigint `>= 0n`
```

katelynsills commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@agoric/nat",
-  "version": "3.0.2-dev.0",
+  "version": "4.0.0-dev.0",
```
Was this done manually? If so, let's not change the version manually and instead use the release instructions in CONTRIBUTING.md. Those release instructions have been altered for `yarn` rather than `npm` in https://github.com/Agoric/nat/pull/144 but may need additional fixes.

katelynsills commented on src/index.js:
```diff
@@ -13,28 +13,64 @@
 }
 
-export default Nat;
+export { isNat, Nat };
```
We should mention the change from default export in NEWS.md if we don't already

erights replied:
Good catch. I completely missed this.

katelynsills commented on test/test.js:
```diff
@@ -1,44 +1,65 @@
+  t.end();
+});
+
+test('isNat examples from the README', t => {
```
I like including this!

erights replied:
;)

katelynsills reviewed (commented):
This PR looks great! There are some ad-hoc changes that were done partially (like including a yarn.lock but not deleting package-lock.json or doing further work in changing to `yarn`) which I covered in more depth in #144, #145, and #148. I'd like to propose that this PR be rebased on those PRs for any tooling changes.

erights commented:
I rebased on top of your changes.

I reverted the manual version number change. I then tried following the instructions in CONTRIBUTING.md and got confused. Should I just go ahead and merge (following your approval) and worry about the version / CONTRIBUTING / release issues after merging?

If so, then PTAL. It's ready for another review. Otherwise I need someone looking over my shoulder for the CONTRIBUTING part.

katelynsills commented:
Yup, I or @michaelfig  can do the release. Thanks so much for rebasing!

katelynsills reviewed (approved):
LGTM!
